File: Cpp/112.cpp
#include<iostream>
#include "Tree.h"

using namespace std;

class Solution {
public:
    bool hasPathSum(TreeNode* root, int sum) {
        return hasSum(root, sum);
    }
    
    bool hasSum(TreeNode* root, int sum){
        if(!root){
            return false;
        }
        
        sum -= root->val;
        if(!root->left && !root->right){
            return sum == 0;
        }
        
        return hasSum(root->left, sum) || hasSum(root->right, sum);
    }
};

int main(){
    
}

File: Python3/112.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        if not root:
            return False
        if not root.left and not root.right and root.val == targetSum:
            return True
        return self.hasPathSum(root.left, targetSum-root.val) or self.hasPathSum(root.right, targetSum-root.val)
